ci_framework/roles/cifmw_cephadm/tasks/dashboard/dashboard.yml

```diff
@@ -0,0 +1,109 @@
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Get ceph_cli
+  ansible.builtin.include_tasks: ceph_cli.yml
+  vars:
+    mount_spec: true
+    sensitive_data: true
+    mount_certs: "{{ (cifmw_cephadm_dashboard_crt | length > 0 and cifmw_cephadm_dashboard_key | length > 0) | bool }}"
+
+- name: Configure the Ceph Dashboard port
+  become: true
+  block:
+    - name: "set the dashboard port ({{ cifmw_cephadm_dashboard_port }})"
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/server_port {{ cifmw_cephadm_dashboard_port }}"
+      changed_when: false
+    - name: "set the dashboard SSL port ({{ cifmw_cephadm_dashboard_port }})"
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl_server_port {{ cifmw_cephadm_dashboard_port }}"
+      run_once: true
+
+- name: disable SSL for dashboard
+  become: true
+  ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl false"
+
+- name: Configure SSL key/cert for the Ceph Dashboard if provided
+  become: true
+  block:
+    - name: enable SSL for dashboard
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl true"
+      run_once: true
+
+    - name: import dashboard certificate file
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} config-key set mgr/dashboard/crt -i {{ cifmw_cephadm_dashboard_crt }}"
+      changed_when: false
+
+    - name: import dashboard certificate key
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} config-key set mgr/dashboard/key -i {{ cifmw_cephadm_dashboard_key }}"
+      changed_when: false
+  when: cifmw_cephadm_dashboard_crt | length > 0 and
+        cifmw_cephadm_dashboard_key | length > 0
+
+- include_tasks: configure_dashboard_backends.yml
+  with_items: "{{ groups['edpm'] }}"
+  vars:
+    dashboard_backend: '{{ item }}'
+
+- name: Restart the Ceph dashboard
+  become: true
+  block:
+    - name: disable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module disable dashboard"
+    - name: enable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module enable dashboard"
+
+- name: Configure Monitoring Stack
+  become: true
+  vars:
+    grafana_host:  "{{ groups['edpm'][0] }}"
+  block:
+    - name: get Grafana instance address
+      ansible.builtin.set_fact:
```

Comment:
So, if we get `|first` here, where we set `grafana_server_addr`, we can remove that filter everywhere in the code below

Reply:
Good catch, thanks